x-pack/plugins/security_solution/common/endpoint/data_loaders/index_endpoint_hosts.ts

```diff
@@ -0,0 +1,312 @@
+    );
+
+    if (enrollFleet) {
+      const { id: appliedPolicyId, name: appliedPolicyName } = hostMetadata.Endpoint.policy.applied;
+
+      // If we don't yet have a "real" policy record, then create it now in ingest (package config)
+      if (!realPolicies[appliedPolicyId]) {
+        const createdPolicies = await indexFleetEndpointPolicy(
+          kbnClient,
+          appliedPolicyName,
+          epmEndpointPackage.version
+        );
+
+        merge(response, createdPolicies);
+
+        // eslint-disable-next-line require-atomic-updates
+        realPolicies[appliedPolicyId] = createdPolicies.integrationPolicies[0];
+      }
+
+      // If we did not yet enroll an agent for this Host, do it now that we have good policy id
+      if (!wasAgentEnrolled) {
+        wasAgentEnrolled = true;
+
+        const indexedAgentResponse = await indexFleetAgentForHost(
+          client,
+          kbnClient,
+          hostMetadata!,
+          realPolicies[appliedPolicyId].policy_id,
+          kibanaVersion
+        );
+
+        enrolledAgent = indexedAgentResponse.agents[0];
+        merge(response, indexedAgentResponse);
+      }
+      // Update the Host metadata record with the ID of the "real" policy along with the enrolled agent id
+      hostMetadata = {
+        ...hostMetadata,
+        elastic: {
+          ...hostMetadata.elastic,
+          agent: {
+            ...hostMetadata.elastic.agent,
+            id: enrolledAgent?.id ?? hostMetadata.elastic.agent.id,
+          },
+        },
+        Endpoint: {
+          ...hostMetadata.Endpoint,
+          policy: {
+            ...hostMetadata.Endpoint.policy,
+            applied: {
+              ...hostMetadata.Endpoint.policy.applied,
+              id: realPolicies[appliedPolicyId].id,
+            },
+          },
+        },
+      };
+
+      // Create some actions for this Host
+      await indexFleetActionsForHost(client, hostMetadata);
+    }
+
+    await client
+      .index({
+        index: metadataIndex,
+        body: hostMetadata,
+        op_type: 'create',
+      })
+      .catch(wrapErrorAndRejectPromise);
+
+    const hostPolicyResponse = generator.generatePolicyResponse({
+      ts: timestamp - timeBetweenDocs * (numDocs - j - 1),
+      policyDataStream: EndpointDocGenerator.createDataStreamFromIndex(policyResponseIndex),
+    });
+
+    await client
+      .index({
+        index: policyResponseIndex,
+        body: hostPolicyResponse,
+        op_type: 'create',
+      })
+      .catch(wrapErrorAndRejectPromise);
+
+    // Clone the hostMetadata and policyResponse document to ensure that no shared state
+    // (as a result of using the generator) is returned across docs.
+    response.hosts.push(cloneDeep(hostMetadata));
+    response.policyResponses.push(cloneDeep(hostPolicyResponse));
+  }
+
+  return response;
+}
+
+const fetchKibanaVersion = async (kbnClient: KbnClient) => {
+  const version = ((await kbnClient.request({
```

Comment:
Do you need a try catch? Kibana could also be down.

Reply:
I don't think its necessary. We need Kibana to be available, so if this call fails, `throw`ing is ok IMO